ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/BigLedgerPeers.hs

```diff
@@ -0,0 +1,223 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Ouroboros.Network.PeerSelection.Governor.BigLedgerPeers
+  ( belowTarget
+  , aboveTarget
+  ) where
+
+import qualified Data.Map.Strict as Map
+import           Data.Set (Set)
+import qualified Data.Set as Set
+
+import           Control.Applicative (Alternative)
+import           Control.Concurrent.JobPool (Job (..))
+import           Control.Exception (SomeException)
+import           Control.Monad.Class.MonadSTM
+import           Control.Monad.Class.MonadTime.SI
+
+import           Ouroboros.Network.PeerSelection.LedgerPeers (IsLedgerPeer (..))
+import           Ouroboros.Network.PeerSelection.Governor.Types
+import           Ouroboros.Network.PeerSelection.PeerAdvertise (PeerAdvertise (..))
+import           Ouroboros.Network.PeerSelection.PeerSharing (PeerSharing (..))
+import qualified Ouroboros.Network.PeerSelection.State.EstablishedPeers as EstablishedPeers
+import qualified Ouroboros.Network.PeerSelection.State.KnownPeers as KnownPeers
+import qualified Ouroboros.Network.PeerSelection.State.LocalRootPeers as LocalRootPeers
+
+
+belowTarget :: (MonadSTM m, Ord peeraddr)
```

Comment:
TODO: think about how to check that the target is achievable, e.g. it's smaller than number of big ledger peers.